lib/packwerk/const_node_inspector.rb

```diff
@@ -28,23 +34,27 @@ def constant_name_from_node(node, ancestors:)
 
     # Only process the root `const` node for namespaced constant references. For example, in the
     # reference `Spam::Eggs::Thing`, we only process the const node associated with `Spam`.
+    sig { params(parent: T.nilable(AST::Node)).returns(T::Boolean) }
     def root_constant?(parent)
       !(parent && Node.constant?(parent))
     end
 
+    sig { params(node: AST::Node, parent: AST::Node).returns(T.nilable(T::Boolean)) }
     def constant_in_module_or_class_definition?(node, parent:)
       parent_name = Node.module_name_from_definition(parent)
       parent_name && parent_name == Node.constant_name(node)
     end
 
+    sig { params(node: AST::Node, ancestors: T::Array[AST::Node]).returns(String) }
     def fully_qualify_constant(node, ancestors:)
       # We're defining a class with this name, in which case the constant is implicitly fully qualified by its
       # enclosing namespace
       name = Node.parent_module_name(ancestors: ancestors)
-      name ||= generate_qualified_constant(node, ancestors)
+      name ||= generate_qualified_constant(node, ancestors: ancestors)
```

Comment:
this is strange. How did this even work before?

Reply:
Is this line covered by tests?

Reply:
I know right? It can't have been covered by tests I guess 🤷‍♂️  - I was quite surprised Sorbet didn't catch it just based on the arguments

Reply:
yeah it's sometimes surprising what sorbet misses.

Reply:
I guess we are never running into a situation where we need to generate a qualified constant because Node.parent_module_name is not returning nil.  Core is also not running into this issue so maybe we can get rid of the method and see what happens?

Reply:
I chased this today:
- `name = Node.parent_module_name(ancestors: ancestors)` is never empty, it will always return `"Object"` or the module name
- This means we never run the `name ||=` line since `name` is never `nil`

I added a sig to `Node.parent_module_name` in https://github.com/Shopify/packwerk/pull/92/commits/d1a5b7d35fcfd432b2bd41278d6e604ac6b5dacf and at this point Sorbet was able to confirm this line is unreachable:
![image](https://user-images.githubusercontent.com/2643026/101872168-b4309b80-3b7c-11eb-8050-dc009ff0069d.png)

Reply:
@tomstuart @wildmaples  You might have a bit more context why we were using generate_qualified_constant just to make sure that we don't have a regression. Given that we aren't reaching this core method and don't have any regressions, we can probably get rid of it. WDYT?

Reply:
I don’t, sorry. This line of code was present when Packwerk was [first extracted](https://github.com/shopify/packwerk-old/commit/7112ab0813f7c4cad859c0519c4baee24894041e#diff-b9878467a38ac6cfae24fc57d7ef5e853dfeb5f5b6873147c648201725ce6c01R37) from Shopify’s core monolith and I haven’t traced it any further back than that. The most likely explanation is that the author was unaware that `RuboCop::AST::Node#parent_module_name` is [incapable of returning `nil`](https://github.com/rubocop-hq/rubocop-ast/blob/5cd306f40e5d5ba4dacf78c698354747cdac7825/lib/rubocop/ast/node.rb#L312-L320) and was trying to be defensive. I agree we should remove it.